tests/sentry/incidents/test_models.py

```diff
@@ -553,3 +557,44 @@ def test_update(self):
         assert AlertRuleActivity.objects.filter(
             alert_rule=self.alert_rule, type=AlertRuleActivityType.UPDATED.value
         ).exists()
+
+
+class CleanExpiredAlertsTest(TestCase):
+    @patch("sentry.incidents.models.delete_snuba_subscription")
+    def test_clean_expired_alerts_active(self, mock_delete_snuba_subscription):
+        alert_rule = self.create_alert_rule(monitor_type=AlertRuleMonitorType.ACTIVATED)
+        subscription = alert_rule.snuba_query.subscriptions.get()
+
+        clean_expired_alerts(subscription)
+        assert mock_delete_snuba_subscription.call_count == 0
```

Comment:
Instead of mocking, would it be better to use `with self.tasks()` and just verify that the subscription hasn't been deleted?